### PR TITLE
Jetpack section in My Site

### LINKED ISSUE #15275: Jetpack Section: My Site Menu Improvements
Project: #15193

### Tasks
- [x] Add a new value to `BlogDetailsSectionCategory` enum
- [x] Create a new `BlogDetailsSection` object for the Jetpack section grouping
- [x] Create a new `BlogDetailsRow` object for the new Jetpack `Settings` action
- [x] Add action to the row that creates and pushes to a new settings screen with 2 rows : Security and Manage connection 
- [x] Check if this change effects Quick Start
- [x] Add tracking event for Jetpack settings

### PR DESCRIPTION
**Pull request by yaelirub**
Fixes #15275
 
This PR groups Stats, Activity log and Jetpack Settings in the Jetpack section. 
Jetpack settings have been extracted from site settings and now exist in both places. 

This PR also renames `JetpackSecuritySettingsViewController` to `JetpackSettingsViewController` 
and groups the the related classes under a new subfolder - Jetpack Settings

New Tracks:
jetpack_settings_viewed
jetpack_manage_connection_viewed
jetpack_disconnect_tapped
jetpack_disconnect_requested - (User selected disconnect in alert)

jetpack_section

To test:
1. Go to My Site
2. See the Jetpack Section 
3. See Jetpack Settings row only if you have a plan 
4. Test all new events are fired correctly 
5. Test it doesn't affect Quick Start 

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by yaelirub** [comment]
@emilylaguna, in addition to the code review and described testing could you please also:
* Help improve the release note 😬 
* Test Quick start is not affected
?
🙏

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -3,6 +3,8 @@
 * [**] Support contact email: fixed issue that prevented non-alpha characters from being entered. [#15210]
 * [*] Support contact information prompt: fixed issue that could cause the app to crash when entering email address. [#15210]
 * [*] Fixed an issue where comments viewed in the Reader would always be italicized.
+* [**] Jetpack Section - Easy access to Jetpack settings under My Site.
+nu
```
Comment (emilylaguna) [inline RELEASE-NOTES.txt]:
```suggestion
* [**] Jetpack Section - Added quick and easy access for all the Jetpack features (Stats, Activity Log, Jetpack and Settings) on the My Site view [#15287]
```

File: WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m [hunk 2]
```diff
@@ -1545,6 +1600,13 @@ - (void)showSettings
     [[QuickStartTourGuide find] visited:QuickStartTourElementBlogDetailNavigation];
 }
 
+-(void)showJetpackSettings
+{
+    [WPAnalytics trackEvent:WPAnalyticsEventJetpackSettingsViewed];
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m]: Should we move this event to `viewDidLoad` in `JetpackSettingsViewController` since this view can be accessed in other areas of the app as well?

File: WordPress/Classes/ViewRelated/Blog/Site Settings/SiteSettingsViewController.m [hunk 3]
```diff
@@ -1333,7 +1333,7 @@ - (void)showJetpackSettingsForBlog:(Blog *)blog
 
     NSParameterAssert(blog);
 
-    JetpackSecuritySettingsViewController *settings = [[JetpackSecuritySettingsViewController alloc] initWithBlog:blog];
+    JetpackSettingsViewController *settings = [[JetpackSettingsViewController alloc] initWithBlog:blog];
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Blog/Site Settings/SiteSettingsViewController.m:1336]: Do we still need the 'Security' and 'Manage Connection' options on the Site Settings?

**Review by emilylaguna — CHANGES_REQUESTED** [review]
tl;dr Works well! Great job! Left a couple comments below.

--- 
### Tested on: 
- iPhone 11 Pro - 14.0
- iPad Air - 12.4
- iPhone 5s 11.4

### Jetpack Section
- Verified the Activity Log option Jetpack section is visible for sites that support Activity Log, Stats, or have Jetpack settings
- Verified the Jetpack Settings does not appear for sites that don't support it
- Verified the Jetpack section is not visible if the site doesn't support Activity Log (meetomattic)
- Verified each item in the section works

### Jetpack Settings
- Verified each option correctly changes the value on WP.com, tested by changing each option and verifying on Calypso
- Verified Manage Connection will disconnect your site

### Quick Start
- Tested each step in Quick Start 
- Found [1 issue](https://github.com/wordpress-mobile/WordPress-iOS/issues/15294) not related to this PR

### Events: Tested and verified the following tracks are being fired

|| Event | When |
|--|--|--|
|✅ | jetpack_settings_viewed | When tapping on the 'Jetpack Settings' option |
|✅| jetpack_manage_connection_viewed | When tapping on the 'Manage connection' option |
|✅| jetpack_disconnect_tapped | When tapping the 'Disconnect' button on Manage Connection |
|✅| jetpack_disconnect_requested | When tapping Disconnect on the confirmation alert |

File: WordPress/Classes/ViewRelated/Blog/Site Settings/SiteSettingsViewController.m (hunk 3 above)
Comment (yaelirub) [reply WordPress/Classes/ViewRelated/Blog/Site Settings/SiteSettingsViewController.m:1336]: As described, i chose to to keep them there for now. I can imagine that in the future we will remove them

File: WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m (hunk 2 above)
Comment (yaelirub) [reply WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m]: Done in cbe18049e8229d0ff786cb82b676e90d0ed22039

**Comment by yaelirub** [comment]
Ready for another look

**Review by emilylaguna — APPROVED** [review]
Looks good to me! 🚢
